core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphqlIndividual.kt

```diff
@@ -0,0 +1,109 @@
+package org.evomaster.core.problem.graphql
+
+import org.evomaster.core.Lazy
+import org.evomaster.core.database.DbAction
+import org.evomaster.core.database.DbActionUtils
+import org.evomaster.core.problem.rest.*
+import org.evomaster.core.search.Action
+import org.evomaster.core.search.Individual
+import org.evomaster.core.search.gene.*
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.tracer.TraceableElement
+import org.evomaster.core.search.tracer.TrackOperator
+
+class GraphqlIndividual (
+        val actions: MutableList<GraphqlAction>,
+        val sampleType: GraphqlSampleType,
+        val dbInitialization: MutableList<DbAction> = mutableListOf(),
+        val usedObjects: GraphqlUsedObjects = GraphqlUsedObjects(),
+        trackOperator: TrackOperator? = null,
+        traces : MutableList<out GraphqlIndividual>? = null
+): Individual(trackOperator, traces) {
+
+    override fun copy(): Individual {
+        return GraphqlIndividual(
+                actions.map { a -> a.copy() as GraphqlAction } as MutableList<GraphqlAction>,
+                sampleType,
+                dbInitialization.map { d -> d.copy() as DbAction } as MutableList<DbAction>,
+                usedObjects.copy(),
+                trackOperator
+        )
+    }
+
+    override fun canMutateStructure(): Boolean {
+        return sampleType == SampleType.RANDOM
+    }
+    override fun seeGenes(filter: GeneFilter): List<out Gene> {
+
+        return when (filter) {
+            GeneFilter.ALL -> dbInitialization.flatMap(DbAction::seeGenes).plus(actions.flatMap(GraphqlAction::seeGenes))
+            GeneFilter.NO_SQL -> actions.flatMap(GraphqlAction::seeGenes)
+            GeneFilter.ONLY_SQL -> dbInitialization.flatMap(DbAction::seeGenes)
+
+        }
+    }
+
+    override fun size() = actions.size
+
+    override fun seeActions(): List<out Action> {
+        return actions
+    }
+
+    override fun seeInitializingActions(): List<Action> {
+        return dbInitialization
+    }
+
+    override fun verifyInitializationActions(): Boolean {
+        return DbActionUtils.verifyActions(dbInitialization.filterIsInstance<DbAction>())
+    }
+
+    override fun repairInitializationActions(randomness: Randomness) {
```

Comment:
this is fine for now.
at a certain point, I will move the DB code in its own module, so it can be re-used.
but, if copy&paste code like this, do write a //TODO comment specifying it will need to be removed/refactored